indexers/vector/ZarrIndexer/__init__.py

```diff
@@ -0,0 +1,63 @@
+from os import path
+from typing import List, Union, Optional
+
+import zarr
+import numpy as np
+
+from jina.executors.indexers.vector import NumpyIndexer
+from jina.helper import cached_property
+
+
+class ZarrIndexer(NumpyIndexer):
+    """Indexing based on Zarr arrays
+    
+    For more information about Zarr, please check
+    https://zarr.readthedocs.io/en/stable/index.html
+    
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        
+    def get_add_handler(self) -> zarr.hierarchy.Group:
+        """Open an existing zarr group file for adding new vectors
+
+        :return: a zarr group file
+        """
+        return zarr.open(store=self.index_abspath, mode='a')
+
+    def get_create_handler(self) -> zarr.hierarchy.Group:
+        """Create a zarr group file for adding new vectors
+
+        :return: a zarr group file
+        """
+        return zarr.open(store=self.index_abspath, mode='w')
+    
+    def add(self, keys: 'np.ndarray', vectors: 'np.ndarray', *args, **kwargs) -> None:
+        self._validate_key_vector_shapes(keys, vectors)
```

Comment:
Is this validated in every addition for other indexers?

Reply:
It is done in the same way in `NumpyIndexer`